src/parameterized_tendencies/gravity_wave_drag/orographic_gravity_wave.jl

```diff
@@ -16,23 +16,31 @@ Not yet included in our codebase
 
 using ClimaCore: InputOutput
 
-orographic_gravity_wave_cache(::Nothing, Y, radius) = NamedTuple()
+orographic_gravity_wave_cache(Y, atmos::AtmosModel) =
+    orographic_gravity_wave_cache(Y, atmos.orographic_gravity_wave)
+
+orographic_gravity_wave_cache(Y, ::Nothing) = (;)
 
 orographic_gravity_wave_tendency!(Yₜ, Y, p, t, ::Nothing) = nothing
 
 include(joinpath(pkgdir(ClimaAtmos), "artifacts", "artifact_funcs.jl"))
 
-function orographic_gravity_wave_cache(ogw::OrographicGravityWave, Y, radius)
+function orographic_gravity_wave_cache(Y, ogw::OrographicGravityWave)
+
+    @assert Spaces.horizontal_space(axes(Y.c)).topology.mesh.domain isa
+            Domains.SphereDomain
```

Comment:
```suggestion
    @assert Spaces.horizontal_space(axes(Y.c)).topology.mesh.domain isa
            Domains.SphereDomain
```
is there a more direct accessor function we can use here?

Reply:
We have `domain`, which takes in a `Mesh`, but it wouldn't be much better I think

Reply:
Ah, bummer. We should probably add something to ClimaCore to hide internal structure a bit more.